### ISSUE TITLE
Verifiable Credentials Endpoint

### ISSUE THREAD

**Issue by MimiaBena** [openid/OpenID4VCI#224]
Hello,
To create a credential, we have two endpoints: one for creating a single credential and another for a batch of credentials. When the wallet receives an access token, how does the wallet know which endpoint to use for a verifiable credential or for a batch of credentials (access Token ise encrypted)?

Any help, pls.

Thanks

**Comment by jogu** [comment]
If the issuer lists both endpoints in its metadata the wallet can use whichever one it prefers. The 'single credential' one is required to be supported by the issuer so the wallet could also decide to always call that one.

**Comment by Sakurann** [comment]
The fact that there are details to be worked out for Batch endpoint is probably orthogonal to the question, and the answer is what Joseph said (just to reiterate):
- it is wallet's choice which of the endpoint to use if the issuer is listing more than one (which also means that the issuer needs to make sure that the same access token is valid at all of the endpoints supported by the issuer)
- credential endpoint is currently mandatory to implement, so that is always a fallback option.

Thanks for the question and please let us know if the question has been answered so that we can resolve the issue :)

**Comment by MimiaBena** [comment]
So, what I understand, for example, is that if the wallet has chosen to use the Endpoint credentials and has entered multiple VCS requests within this endpoint, it must process all the VCS. The batch credentials are optional. In my example, the wallet receives the access token, but it doesn't know how much data I have placed inside. I hope I was clear in my explanation.

Thank you!

**Comment by jogu** [comment]
If you still have concerns, I think it would be helpful if you described the entire flow you're thinking of in your example. I'm not sure if you're asking about credential offers, pre authorised codes, if authorization_details was used or not, etc. If you lay out the sequence of what happens in the first steps of your example then we should be able to explain what the wallet is expected to do next.

**Comment by MimiaBena** [comment]
Sure. So, the flow:
I created a basic authorization flow with a score, and I didn't use authorization_details. After that, I chose multiple data (multiple VCS requests). Then, I constructed my access token. The wallet receives the access token but doesn't know what I put in the access token. At this point, I didn't know how the wallet would choose the correct endpoint (the batch or the credential endpoint – initially, I thought this endpoint was only for creating a single credential). I hope I have been clear in my explanation.
